galley/pkg/config/analysis/analyzers/injection/injection.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package injection
+
+import (
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+	"istio.io/istio/galley/pkg/config/processor/metadata"
+	"istio.io/istio/galley/pkg/config/resource"
+)
+
+// Analyzer checks conditions related to Istio injection
+type Analyzer struct{}
+
+var _ analysis.Analyzer = &Analyzer{}
+
+// Name implements Analyzer
+func (s *Analyzer) Name() string {
+	return "injection.Analyzer"
+}
+
+// We assume that enablement is via an istio-injection=enabled namespace label
+// In theory, there can be alternatives using Mutatingwebhookconfiguration, but they're very uncommon
+// See https://istio.io/docs/ops/troubleshooting/injection/ for more info.
```

Comment:
knative docs use `enableNamespacesByDefault=true` which translates to the istio-injection=disabled method. See https://knative.dev/docs/install/installing-istio/#installing-istio-with-sidecar-injection

Reply:
Is there some resource we can look at that will give us the default? Is that via `mutatingwebhookconfigurations`? Right now I moved away from using that to reduce complexity.

Reply:
You can find it in the values field of the injector config map. 

```bash
$ kubectl -n istio-system get cm istio-sidecar-injector -o jsonpath='{.data.values}' \
        | jq '.sidecarInjectorWebhook.enableNamespacesByDefault'
false
```

Reply:
Thanks. Added a tracking comment to tackle this case later